### PR TITLE
Fix 02981_vertical_merges_memory_usage with SharedMergeTree

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 2d42054845c123895b77b2be3166f7f3638c4992 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60028/2d42054845c123895b77b2be3166f7f3638c4992/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60028/2d42054845c123895b77b2be3166f7f3638c4992/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/02981_vertical_merges_memory_usage.sql [hunk 1]
```diff
@@ -14,7 +14,7 @@ SETTINGS
     merge_max_block_size_bytes = '10M';
 
 INSERT INTO t_vertical_merge_memory SELECT number, arrayMap(x -> repeat('a', 50), range(1000)) FROM numbers(3000);
-INSERT INTO t_vertical_merge_memory SELECT number, arrayMap(x -> repeat('a', 50), range(1000)) FROM numbers(3000);
+INSERT INTO t_vertical_merge_memory SELECT number, arrayMap(x -> repeat('a', 50), range(1000)) FROM numbers(3001);
```
Comment (nikitamikhaylov) [inline tests/queries/0_stateless/02981_vertical_merges_memory_usage.sql:17]: 🤔

Comment (Algunenano) [reply tests/queries/0_stateless/02981_vertical_merges_memory_usage.sql:17]: Deduplication, which is off with normal MergeTree by default

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/02981_vertical_merges_memory_usage.sql (hunk 1 above)
Comment (Algunenano) [reply tests/queries/0_stateless/02981_vertical_merges_memory_usage.sql:19]: Maybe we should randomize `non_replicated_deduplication_window`, but in another PR.

**Pull request merged**